diff --git a/select/index.d.ts b/select/index.d.ts
index a31edd4e5094a66bf99fec28c6d7568330ae677c..432d4acfb18e417355026615cb2326a433f6f8e4 100644
--- a/select/index.d.ts
+++ b/select/index.d.ts
@@ -14,7 +14,7 @@ import type { InputIconPassThroughOptions } from 'primevue/inputicon';
 import type { InputTextPassThroughOptions } from 'primevue/inputtext';
 import type { PassThroughOptions } from 'primevue/passthrough';
 import type { VirtualScrollerItemOptions, VirtualScrollerPassThroughOptionType, VirtualScrollerProps } from 'primevue/virtualscroller';
-import { TransitionProps, VNode } from 'vue';
+import { AllowedComponentProps, ComponentCustomProps, TransitionProps, VNode, VNodeProps } from 'vue';
 
 export declare type SelectPassThroughOptionType<T = any> = SelectPassThroughAttributes | ((options: SelectPassThroughMethodOptions<T>) => SelectPassThroughAttributes | string) | string | null | undefined;
 
@@ -290,7 +290,7 @@ export interface SelectContext {
 /**
  * Defines valid properties in Select component.
  */
-export interface SelectProps {
+export interface SelectProps<T = any> {
     /**
      * Value of the component.
      */
@@ -306,27 +306,27 @@ export interface SelectProps {
     /**
      * An array of select items to display as the available options.
      */
-    options?: any[];
+    options?: T[];
     /**
      * Property name or getter function to use as the label of an option.
      */
-    optionLabel?: string | ((data: any) => string) | undefined;
+    optionLabel?: string | ((data: T) => string) | undefined;
     /**
      * Property name or getter function to use as the value of an option, defaults to the option itself when not defined.
      */
-    optionValue?: string | ((data: any) => any) | undefined;
+    optionValue?: string | ((data: T) => any) | undefined;
     /**
      * Property name or getter function to use as the disabled flag of an option, defaults to false when not defined.
      */
-    optionDisabled?: string | ((data: any) => boolean) | undefined;
+    optionDisabled?: string | ((data: T) => boolean) | undefined;
     /**
      * Property name or getter function to use as the label of an option group.
      */
-    optionGroupLabel?: string | ((data: any) => string) | undefined;
+    optionGroupLabel?: string | ((data: T) => string) | undefined;
     /**
      * Property name or getter function that refers to the children options of option group.
      */
-    optionGroupChildren?: string | ((data: any) => any[]) | undefined;
+    optionGroupChildren?: string | ((data: T) => any[]) | undefined;
     /**
      * Height of the viewport, a scrollbar is defined if height of list exceeds this value.
      * @defaultValue 14rem
@@ -576,7 +576,7 @@ export interface SelectProps {
 /**
  * Defines valid slots in Select component.
  */
-export interface SelectSlots {
+export interface SelectSlots<T = any> {
     /**
      * Custom value template.
      * @param {Object} scope - value slot's params.
@@ -603,7 +603,7 @@ export interface SelectSlots {
         /**
          * Displayed options
          */
-        options: any[];
+        options: T[];
     }): VNode[];
     /**
      * Custom footer template of panel.
@@ -617,7 +617,7 @@ export interface SelectSlots {
         /**
          * Displayed options
          */
-        options: any[];
+        options: T[];
     }): VNode[];
     /**
      * Custom option template.
@@ -627,7 +627,7 @@ export interface SelectSlots {
         /**
          * Option instance
          */
-        option: any;
+        option: T;
         /**
          * Selection state
          */
@@ -645,7 +645,7 @@ export interface SelectSlots {
         /**
          * Option instance
          */
-        option: any;
+        option: T;
         /**
          * Index of the option
          */
@@ -692,7 +692,7 @@ export interface SelectSlots {
         /**
          * Options of the loader items for virtualscroller
          */
-        options: any[];
+        options: T[];
     }): VNode[];
     /**
      * Custom clear icon template.
@@ -822,11 +822,19 @@ export interface SelectMethods {
  * @group Component
  *
  */
-declare const Select: DefineComponent<SelectProps, SelectSlots, SelectEmits, SelectMethods>;
+declare const Select: {
+    new <T = any>(
+        props: SelectProps<T>
+    ): {
+        $props: SelectProps<T> & VNodeProps & AllowedComponentProps & ComponentCustomProps;
+        $slots: SelectSlots<T>;
+        $emit: EmitFn<SelectEmitsOptions>;
+    } & SelectMethods;
+};
 
 declare module 'vue' {
     export interface GlobalComponents {
-        Select: DefineComponent<SelectProps, SelectSlots, SelectEmits, SelectMethods>;
+        Select: typeof Select;
     }
 }
 
